### PR TITLE
Update FoundationICU to 0.0.4

### PR DESCRIPTION
**Pull request by iCharlesHu**
This PR updates our FoundationICU dependency to `0.0.4` (corresponds to `ICU 72.1`, the same version shipped in macOS Sonoma).

### CODE REVIEW

**Comment by iCharlesHu** [comment]
@swift-ci please test

**Review by itingliu — APPROVED** [review]
awesome!

**Comment by itingliu** [comment]
Linux CI shows this error that looks legitimate
```
[160/548] Compiling ICUI18N rbt.cpp
/build/swift-foundation/.build/checkouts/swift-foundation-icu/icuSources/i18n/smpdtfmt.cpp:85:10: fatal error: 'os/log.h' file not found
   85 | #include <os/log.h>
      |          ^~~~~~~~~~
1 error generated.
```

**Comment by iCharlesHu** [comment]
@swift-ci please test

**Review by itingliu — APPROVED** [review]
LGTM as long as CI is happy. Please merge when ready 🙏 🙇

File: Sources/TestSupport/TestSupport.swift [hunk 1]
```diff
@@ -203,3 +203,27 @@ public typealias ComparableComparator = FoundationEssentials.ComparableComparato
 public typealias ComparisonResult = FoundationEssentials.ComparisonResult
 
 #endif // FOUNDATION_FRAMEWORK
+
+/// ICU uses `\u{202f}` and `\u{020f}` interchangeably
```
Comment (jmschonfeld) [inline Sources/TestSupport/TestSupport.swift:207]: Very minor nitpick: I think you mean't U+0020 here instead of U+020F 🙂

**Pull request merged**